File: shared/src/commonMain/kotlin/com/tom/paperless/domain/models/uiStates/ItemDetailUiState.kt
package com.tom.paperless.domain.models.uiStates

import com.tom.paperless.domain.models.NfcTaggable

data class ItemDetailUiState(
    val item: NfcTaggable? = null,
    val isLoading: Boolean = false,
    val isSaving: Boolean = false,
    val isDeleting: Boolean = false,
    val errorMessage: String? = null,
    val operationSucceeded: Boolean = false,

    val isDirty: Boolean = false,
    val isEditing: Boolean = false
) {
    companion object {
        fun empty() = ItemDetailUiState()
    }
}

File: iosApp/PaperLess/Application/Ultility/Extensions/ItemDetailUiState.swift
//
//  ItemDetailUiState.swift
//  PaperLess
//
//  Created by Tom Salih on 28.09.25.
//

import Shared

extension ItemDetailUiState {
    static func empty() -> ItemDetailUiState {
        ItemDetailUiState(
            item: nil,
            isLoading: false,
            isSaving: false,
            isDeleting: false,
            errorMessage: nil,
            operationSucceeded: false,
            isDirty: false,
            isEditing: false
        )
    }
}
